File: AFMobile/AFAndroid/app/src/main/java/cz/cvut/fel/matyapav/afandroid/builders/skins/DefaultSkin.java
package cz.cvut.fel.matyapav.afandroid.builders.skins;

import android.content.Context;
import android.graphics.Color;
import android.graphics.Typeface;
import android.view.Gravity;
import android.view.ViewGroup;
import android.widget.AbsListView;

/**
 * Created by Pavel on 24.02.2016.
 */
public class DefaultSkin implements Skin {

    private Context context;

    public DefaultSkin(Context context) {
        this.context = context;
    }

    public Context getContext() {
        return context;
    }

    public int convertDpToPixels(int dps, Context context){
        final float scale = context.getResources().getDisplayMetrics().density;
        return (int) (dps * scale + 0.5f);
    }

    /*******************COMMON*******************/
    @Override
    public ViewGroup.LayoutParams getTopLayoutParams() {
        return new ViewGroup.LayoutParams(ViewGroup.LayoutParams.WRAP_CONTENT, ViewGroup.LayoutParams.WRAP_CONTENT);
    }

    @Override
    public int getComponentMarginBottom() {
        return 0;
    }

    @Override
    public int getComponentMarginLeft() {
        return 0;
    }

    @Override
    public int getComponentMarginRight() {
        return 0;
    }

    @Override
    public int getComponentMarginTop() {
        return 0;
    }

    /*******************FORM*******************/
    @Override
    public int getLabelColor() {
        return Color.BLACK;
    }

    @Override
    public int getFieldColor() {
        return Color.BLACK;
    }

    @Override
    public int getValidationColor() {
        return Color.RED;
    }

    @Override
    public int getInputWidth() {
        return convertDpToPixels(200, getContext());
    }

    @Override
    public Typeface getValidationFont() {
        return Typeface.DEFAULT;
    }

    @Override
    public Typeface getFieldFont() {
        return Typeface.DEFAULT;
    }

    @Override
    public Typeface getLabelFont() {
        return Typeface.DEFAULT_BOLD;
    }

    @Override
    public int getLabelWidth() {
        return convertDpToPixels(125, getContext());
    }

    @Override
    public int getLabelHeight() {
        return ViewGroup.LayoutParams.MATCH_PARENT;
    }


    /*******************TABLE*******************/
    @Override
    public int getTableHeight() {
        return convertDpToPixels(200, getContext());
    }

    @Override
    public int getHeaderRowHeight() {
        return convertDpToPixels(60, getContext()); //with padding
    }

    @Override
    public int getContentRowHeight() {
        return convertDpToPixels(50, getContext());
    }

    @Override
    public int getHeaderRowBackgroundColor() {
        return Color.GREEN;
    }

    @Override
    public int getHeaderRowTextColor() {
        return Color.BLACK;
    }

    @Override
    public int getContentBackgroundColor() {
        return Color.WHITE;
    }

    @Override
    public int getContentTextColor() {
        return Color.GRAY;
    }

    @Override
    public int getBorderColor() {
        return Color.LTGRAY;
    }

    @Override
    public int getCellPaddingLeft() {
        return convertDpToPixels(5, getContext());
    }

    @Override
    public int getCellPaddingRight() {
        return convertDpToPixels(5, getContext());
    }

    @Override
    public int getCellPaddingTop() {
        return convertDpToPixels(5, getContext());
    }

    @Override
    public int getCellPaddingBottom() {
        return convertDpToPixels(5, getContext());
    }

    @Override
    public int getHeaderRowGravity() {
        return Gravity.CENTER_VERTICAL | Gravity.CENTER_HORIZONTAL;
    }

    @Override
    public int getContentGravity() {
        return Gravity.CENTER_VERTICAL | Gravity.CENTER_HORIZONTAL;
    }

    @Override
    public int getBorderWidth() {
        return convertDpToPixels(1, getContext());
    }

    /********************************lists****************************/
    @Override
    public int getListWidth() {
        return AbsListView.LayoutParams.MATCH_PARENT;
    }

    @Override
    public int getListHeight() {
        return convertDpToPixels(200, getContext());
    }

    @Override
    public int getListItemBackgroundColor() {
        return Color.TRANSPARENT;
    }

    @Override
    public int getListItemNameColor() {
        return Color.BLACK;
    }

    @Override
    public int getListItemTextColor() {
        return Color.GRAY;
    }

    @Override
    public Typeface getListItemNameFont() {
        return Typeface.DEFAULT_BOLD;
    }

    @Override
    public Typeface getListItemTextFont() {
        return Typeface.DEFAULT;
    }

    @Override
    public int getListItemNameSize() {
        return 16;
    }

    @Override
    public int getListItemsTextSize() {
        return 10;
    }

    @Override
    public boolean isListItemNameLabelVisible() {
        return true;
    }

    @Override
    public boolean isListItemTextLabelsVisible() {
        return true;
    }

    @Override
    public boolean isListScrollBarAlwaysVisible() {
        return true;
    }

    @Override
    public int getListItemTextPaddingLeft() {
        return convertDpToPixels(10,getContext());
    }

    @Override
    public int getListItemTextPaddingRight() {
        return convertDpToPixels(5,getContext());
    }

    @Override
    public int getListItemTextPaddingTop() {
        return 0;
    }

    @Override
    public int getListItemTextPaddingBottom() {
        return 0;
    }

    @Override
    public int getListItemNamePaddingLeft() {
        return convertDpToPixels(10,getContext());
    }

    @Override
    public int getListItemNamePaddingRight() {
        return 0;
    }

    @Override
    public int getListItemNamePaddingTop() {
        return convertDpToPixels(5,getContext());
    }

    @Override
    public int getListItemNamePaddingBottom() {
        return 0;
    }

    @Override
    public int getListContentWidth() {
        return AbsListView.LayoutParams.MATCH_PARENT;
    }

    @Override
    public int getListBorderColor() {
        return Color.LTGRAY;
    }

    @Override
    public float getListBorderWidth() {
        return 10;
    }
}

File: AFMobile/AFWinPhone/AFWinPhone/builders/skins/DefaultSkin.cs
using Windows.UI;
using Windows.UI.Xaml;
using Windows.UI.Xaml.Media;

namespace AFWinPhone.builders.skins
{
    public class DefaultSkin : Skin
    {
        public virtual int getComponentMarginBottom()
        {
            return 20;
        }

        public virtual int getComponentMarginLeft()
        {
            return 0;
        }

        public virtual int getComponentMarginRight()
        {
            return 0;
        }

        public virtual int getComponentMarginTop()
        {
            return 0;
        }

        public virtual Color getFieldColor()
        {
            return Colors.Black;
        }

        public virtual FontFamily getFieldFont()
        {
            return new FontFamily("Segoe UI");
        }

        public virtual int getFieldFontSize()
        {
            return 18;
        }

        public virtual HorizontalAlignment getInputHorizontalAlignment()
        {
            return HorizontalAlignment.Stretch;
        }

        public virtual int getInputWidth()
        {
            return 200;
        }

        public virtual Color getLabelColor()
        {
            if (Application.Current.RequestedTheme.Equals(ApplicationTheme.Dark))
            {
                return Colors.White;
            }
            else
            {
                return Colors.Black;
            }
        }

        public virtual FontFamily getLabelFont()
        {
            return new FontFamily("Segoe UI");
        }

        public virtual int getLabelFontSize()
        {
            return 18;
        }

        public virtual int getLabelHeight()
        {
            return -1; //see label verticall alignement
        }

        public virtual HorizontalAlignment getLabelHorizontalAlignment()
        {
            return HorizontalAlignment.Left;
        }

        public virtual VerticalAlignment getLabelVerticalAlignment()
        {
            return VerticalAlignment.Stretch;
        }

        public virtual int getLabelWidth()
        {
            return 125;
        }

        public HorizontalAlignment getListContentHorizontalAlignment()
        {
            return HorizontalAlignment.Stretch;
        }

        public virtual Color getListBorderColor()
        {
            if (Application.Current.RequestedTheme.Equals(ApplicationTheme.Dark))
            {
                return Colors.White;
            }
            else
            {
                return Colors.Black;
            }
        }

        public virtual float getListBorderWidth()
        {
            return 1;
        }

        public virtual int getListContentWidth()
        {
            return -1; //see ListContentHorizontalAlignment
        }

        public virtual int getListHeight()
        {
            return 200;
        }

        public virtual HorizontalAlignment getListHorizontalAlignment()
        {
            return HorizontalAlignment.Stretch;
        }

        public virtual Color getListItemBackgroundColor()
        {
            return Colors.Transparent;
        }

        public virtual Color getListItemNameColor()
        {
            return Colors.LightSkyBlue;
        }

        public virtual FontFamily getListItemNameFont()
        {
            return new FontFamily("Segoe UI");
        }

        public virtual int getListItemNamePaddingBottom()
        {
            return 0;
        }

        public virtual int getListItemNamePaddingLeft()
        {
            return 10;
        }

        public virtual int getListItemNamePaddingRight()
        {
            return 0;
        }

        public virtual int getListItemNamePaddingTop()
        {
            return 0;
        }

        public virtual int getListItemNameSize()
        {
            return 20;
        }

        public virtual int getListItemsTextSize()
        {
            return 16;
        }

        public virtual Color getListItemTextColor()
        {
            if (Application.Current.RequestedTheme.Equals(ApplicationTheme.Dark))
            {
                return Colors.White;
            }
            else
            {
                return Colors.Black;
            }
        }

        public virtual FontFamily getListItemTextFont()
        {
            return new FontFamily("Segoe UI");
        }

        public virtual int getListItemTextPaddingBottom()
        {
            return 3;
        }

        public virtual int getListItemTextPaddingLeft()
        {
            return 10;
        }

        public virtual int getListItemTextPaddingRight()
        {
            return 5;
        }

        public virtual int getListItemTextPaddingTop()
        {
            return 0;
        }

        public virtual VerticalAlignment getListVerticalAlignment()
        {
            return VerticalAlignment.Stretch;
        }

        public virtual int getListWidth()
        {
            return -1; //see ListHorizontal
        }

        public virtual Color getValidationColor()
        {
            return Colors.Red;
        }

        public virtual FontFamily getValidationFont()
        {
            return new FontFamily("Segoe UI");
        }

        public virtual int getValidationFontSize()
        {
            return 18;
        }

        public virtual bool isListItemNameLabelVisible()
        {
            return true;
        }

        public virtual bool isListItemTextLabelsVisible()
        {
            return true;
        }

        public virtual bool isListScrollBarAlwaysVisible()
        {
            return true;
        }
    }
}
